### DISCUSSION TITLE
How to evaluate js and get result for webview window in tauri 2.0

### DISCUSSION THREAD

**Discussion by crackleeeessyp** [tauri-apps/wry#1427] [Q&A]
Seems in tauri 2.0, webview window only have eval method, which cannot get the result from js?

```rust
  .on_page_load(move |window, _payload| {
      println!("Page loaded on window '{}'", window.label());
      // how to get result from eval?
      window.eval("console.log('hello')");
  })
```

![image](https://github.com/user-attachments/assets/9d2b85ae-850b-4c05-b6b2-7657c11bf9db)

**Comment by FabianLars** [comment]
wrong repo? In tauri you'll have to extend the eval code with something that calls a tauri api at the end (command or event whatever) if you want to return something.

I think it would be okay to expose https://docs.rs/wry/latest/wry/struct.WebView.html#method.evaluate_script_with_callback in tauri too but there are no plans at the moment afaik
